src/Illuminate/Filesystem/FilesystemAdapter.php

```diff
@@ -631,7 +631,7 @@ protected function replaceBaseUrl($uri, $url)
     {
         $parsed = parse_url($url);
 
-        return $uri->withScheme($parsed['scheme'])->withHost($parsed['host']);
+        return $uri->withScheme($parsed['scheme'])->withHost($parsed['host'])->withPort($parsed['port']);
```

Comment:
This is going to throw an error because $parsed['port'] won't exist in the array for a url without a port. (Like https://laravel.com)

Reply:
Ah i see. It will give a warning `<warning>PHP Notice:  Undefined index: port in ....` when the port is not specified.

My solution would be something like

```php
return $uri->withScheme($parsed['scheme'])->withHost($parsed['host'])->withPort(@$parsed['port']);
// or
return $uri->withScheme($parsed['scheme'])->withHost($parsed['host'])->withPort($parsed['port'] ?? null);
```

Or what is the best solution in laravel way?

Reply:
Would do the second:
```
return $uri->withScheme($parsed['scheme'])->withHost($parsed['host'])->withPort($parsed['port'] ?? null);
```

Reply:
Okay.

Extra note that port can be nullable
https://github.com/php-fig/http-message/blob/efd67d1dc14a7ef4fc4e518e7dee91c271d524e4/src/UriInterface.php#L232-L240